tests/kernel/mem_protect/syscalls/src/main.c

```diff
@@ -169,9 +169,15 @@ void test_string_nlen(void)
 	zassert_equal(err, 0, "user string faulted");
 	zassert_equal(ret, strlen(user_string), "incorrect length returned");
 
+	/* Skip this scenario for nsim_sem emulated board, unfortunately
+	 * the emulator doesn't set up memory as specified in DTS and poking
+	 * this address doesn't fault
```

Comment:
Is this a DTS bug or an emulator bug?  The former we can fix, the latter we should probably report.

No complaints about this as a workaround though.

Reply:
it's the emulator, I briefly talked to @ruuddw about it in the linked GH issue #22561 

I'm planning on leaving that bug open actually, I'll remove the "fixes" tag.

With this patch, all tests pass in sanitycheck for nsim_sem and nsim_em, which gives us the opportunity to get the nsim emulator onto the CI image and help stop the constant bit-rot in the ARC port.

Reply:
@andyross that's not a bug, but feature of the nSIM :)
In a stand-alone simulator we don't really emulate memory thus this problem. But there seem to be some tricks we're going to try to work-around that behavior. If if ever works out we'll revert this fix.